src/realm/db.cpp

```diff
@@ -1747,19 +1760,29 @@ void DB::AsyncCommitHelper::main()
                 m_has_write_mutex = false;
                 continue;
             }
+            else if (m_pending_mx_release_cb) {
+                m_db->do_end_write();
+                std::function<void()> cb = m_pending_mx_release_cb.value();
+                m_pending_mx_release_cb.reset();
+                lg.unlock();
+                cb();
```

Comment:
I think all the invocations of the callbacks need to be surrounded by try-catch? Not sure what you think should happen if they throw, but the async commit helper thread probably shouldn't be killed.